wiring/src/spark_wiring_fuel.cpp

```diff
@@ -95,17 +96,25 @@ namespace particle { namespace detail {
     }
 
     // Converts SOC_REGISTER reading to state of charge of the cell as a percentage
-    float _getSoC(byte MSB, byte LSB) {
+    float _getSoC(byte MSB, byte LSB, byte soc_bits_precision) {
+        float soc_percent = 0;
+
+        // Maxim ModelGauge doc only mentions 18 and 19 bit
         // MSB is the whole number
         // LSB is the decimal, resolution in units 1/256%
-        float decimal = LSB / 256.0;
-        return MSB + decimal;
+        if (soc_bits_precision == 19) {
+            soc_percent = (((uint32_t)MSB << 8) + LSB) / 512.0; // per datasheet
```

Comment:
Nit: use 512.0`f` to avoid potentially compiling in any double-precision floating point routines.